packages/dds/tree/docs/data-model/README.md

```diff
@@ -0,0 +1,166 @@
+However, JavaScript's built-in `JSON.parse()` does not.
+Therefore, we do not currently see a need to preserve arbitrary precision numbers in M1.
+
+##### Reserved Key Space
+While we require an injective function from the JSON domain to the SharedTree data model, the SharedTree data model is a superset of JSON.
+Therefore, projecting arbitrary SharedTree data to JSON requires tradeoffs due to the fact that the SharedTree data model is a superset of JSON.
+One potential tradeoff is the use of reserved key space to preserve node ids, types, and augmentations.
+
+### Durable References
+
+The data model must support creating inexpensive durable references to nodes within the tree.
+This includes scenarios like creating "share link" URLs or building graph-like relationships within the tree.
+
+### Schema
+
+The data model must encode sufficient information that the system can efficiently layer a schema-on-write policy on top of the underlying data model (see #9282).
+
+### Augmentation
+
+The data model must allow a subset of collaborators to unobtrusively attach extra-schema data to the tree in a way that is ignored (but preserved) by clients that are unaware of these augmentations.
+
+## Model
+
+### Node
+
+In the SharedTree data model, each addressable piece of data is represented as a tree ***node***.
+There is a single node at the root of the SharedTree that can not be removed or replaced.
+
+<figure align="center">
+  <img src="./assets/root.drawio.svg" alt=""/>
+  <figcaption>Figure: Implicit root node</figcaption>
+</figure>
+
+### Value
+
+Each tree node has an optional ***value***.
+Values are used to store scalar data, such as numbers and booleans.
+
+<figure align="center">
+  <img src="./assets/scalar-nodes.drawio.svg" alt=""/>
+  <figcaption>Figure: Nodes with values</figcaption>
+</figure>
+
+From the perspective of the SharedTree data model, values are opaque.
+The only tree operation that affects a node's value is 'setValue', which overwrites a node's value with a new opaque value.
+
+From the perspective of the broader Fluid system, node values are [*serializable*](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/datastore-definitions/src/serializable.ts)
+and are traversed by Fluid services such as garbage collection.
+
+### Field
+
+Each tree node has a set of zero or more fields.
+Fields are used to model map-like composite type, where each  ***field*** represents one relationship between a parent node and an ordered ***sequence*** of one or more children.
+
+<figure align="center">
+  <img src="./assets/children.drawio.svg" alt=""/>
+  <figcaption>Figure: A parent node with two fields</figcaption>
+</figure>
+
+The fields of a node are distinguished by a ***key***.
+Fields of the same node are distinguished by a field key.
+From the data model's perspective, field keys are opaque.
```

Comment:
Once again, I'm not sure what opaque means here. Certainly some operations need to check the keys, e.g. schema validation which is mentioned just below. Is this saying something like "generic operations on the data model check keys for equality but do not actually consider the value of the key", or something?

Reply:
I think it means we're treating the keys as arbitrary. The domain model does not precribe what they ought to be or what a given key might mean.

Reply:
So we're not putting requirements on it like "it must be a string" for example - the only requirement is that "field keys are things which can be checked for sameness against each other". That makes sense. It sounds like this is trying to convey the same requirements as the `K` in a `Map<K, V>`, but in a language-agnostic way. Now I'm reading "opaque" to mean "pretend the data model is a templated/generic type and pretend the opaque thing's type is a template/generic parameter" and I think that's more or less right. But I can't think of a good way to articulate that without using programming languages as analogies.

Reply:
Changed to 'opaque byte sequence'.  The intent is to convey that the data model understands that there is a comparable string of bytes, but otherwise assigns no special meaning to the key content.  Specifically, keys have no semantic meaning at this layer.